iosApp/iosAppTests/iosAppTests.swift

```diff
@@ -101,6 +101,40 @@ class iosAppTests: XCTestCase {
         contentController.disconnectMessenger()
     }
 
+    func testConnectionClosed() {
+        // Pull the deployment for use later.
+        var deployment: Deployment?
+        do {
+            deployment = try Deployment()
+        } catch {
+            XCTFail("\(error.localizedDescription)")
+        }
+        guard let deployment = deployment else {
+            XCTFail("Failed to pull the deployment config.")
+            return
+        }
+
+        // Create 4 new content controllers. We'll use these to trigger a connection closed event.
+        // Right now, there's a max number of 3 open sesssions that use the same token.
+        // We'll open 4 and confirm that an error occurs in the fourth attempt on the oldest client.
+        var controllers = [TestContentController]()
+        for _ in 1...4 {
+            controllers.append(TestContentController(deployment: deployment))
+        }
+        controllers[0].connectionClosed = XCTestExpectation(description: "Wait for the ConnectionClosedEvent")
+        for controller in controllers {
+            controller.startMessengerConnection()
+            delay(3)
+        }
+        let result = XCTWaiter().wait(for: [controllers[0].connectionClosed!], timeout: 30)
+        XCTAssertEqual(result, .completed, "Did not receive a Connection Closed event.")
+
+        // Cleanup. Also verifies that all of the other clients are still connected due to an error being thrown if we disconnect while not being connected.
+        for x in 1...3 {
```

Comment:
Just wondering if `x` is common identifier for iterator placeholder on swift? I know in most languages the convention is to use `i`.

Reply:
It might even be more conventional in Swift to be more verbose with iterator names, e.g.

```
for index in 1...3 { … }
```

as those are the examples used in the Swift language docs.

It might also be a good idea to use the array's count to construct the range to mitigate a potential out of bounds error:

```
for index in 1..<controllers.count {
    controllers[index].disconnectMessenger()
}
```

Reply:
Missed this comment before I merged. I'll add this in on my next test update.